src/System.Net.WebProxy/src/System/Net/WebProxy.cs

```diff
@@ -200,7 +200,7 @@ public bool IsBypassed(Uri host)
 
             return
                 Address == null ||
-                host.IsLoopback ||
+                (BypassProxyOnLocal && host.IsLoopback) ||
                 (BypassProxyOnLocal && IsLocal(host)) ||
```

Comment:
Rather than checking BypassProxyOnLocal twice, it would be better to either do:
```C#
(BypassProxyOnLocal && (host.IsLoopback || IsLocal(host)))
```
or just add the host.IsLoopback as a check in `IsLocal`.

Reply:
Moved the loopback check to IsLocal and ignored the failing test on .NET framework.